### ISSUE TITLE
[Bug]: When connect wallet is rejected from wallet, do not auto-set wallets again

### ISSUE THREAD

**Issue by Adamj1232** [blocknative/web3-onboard#928]
### Current Behavior
Sumbitted through discord by [L03TJ3](https://gist.github.com/L03TJ3)
Where I try to connect a users previously connected wallet.
(which is just based on the docs explaining connecting a existing wallet).

Steps to reproduce:
1. Connect a wallet to the app and store the connected wallet data in local-storage
2. Close the app-window
3. Clear connected-sites in metamask
4. Revisit app
5. Onboard tries to connect based on local-storage
6. Connection request pops-up in metamask
7. Reject
8. Onboard modal shows connection rejected

The issue?

1. Clicking 'back to wallets' retriggers auto-select (where a user might want to use a different wallet to connect?)
2. No event is triggered to catch any rejection error/event, and the connect promise is only resolved when closing the modal (so a user is forced to close the modal, with no explanation)

basically there is no action that can be taken upon user rejecting an auto-select request

### Expected Behavior

If connection is rejected by the user in the wallet and the 'Back to Wallets' button is clicked there is not an attempt to auto connect wallets 

### Steps To Reproduce

Steps to reproduce:
1. Connect a wallet to the app and store the connected wallet data in local-storage
2. Close the app-window
3. Clear connected-sites in metamask
4. Revisit app
5. Onboard tries to connect based on local-storage
6. Connection request pops-up in metamask
7. Reject
8. Onboard modal shows connection rejected

### Onboard Version

Core - 2.2.0

### Node Version

16.13.2

**Comment by Adamj1232** [comment]
Tracking this issue with PR [here](https://github.com/blocknative/web3-onboard/pull/933)

**Comment by Adamj1232** [comment]
[L03TJ3](https://gist.github.com/L03TJ3) This update has been merged and is live now - thank you for reporting!
